functions.lisp

```diff
@@ -3,6 +3,44 @@
 
 (in-package #:njson)
 
+;; FIXME: CL pathname parsing on SBCL+Linux parses #p"/i\\j" as
+;; #p"/ij", which breaks one of JSON Pointer RFC example. It's bad,
+;; but avoidable with e.g. `make-pathname'.
+(defun parse-pointer-pathname (pointer-pathname)
+  "Parse POINTER-PATHNAME per JSON Pointer rules (https://www.rfc-editor.org/rfc/rfc6901).
+Only supports JSON string representation, not the URL one."
+  (flet ((resolve-tildes (string)
+           (uiop:frob-substrings
+            string '("~1" "~0")
+            (lambda (match frob)
+              (funcall frob (case (elt match 1)
+                              (#\1 "/")
+                              (#\0 "~"))))))
+         (read-until (char stream)
+           "Read from STREAM until encountering CHAR.
+CHAR is left unread on STREAM after returning."
+           (coerce (loop for peeked = (peek-char nil stream nil nil)
+                         until (or (eql char peeked)
+                                   (null peeked))
+                         collect (read-char stream nil nil))
+                   'string))
+         (parse-if-number (string)
+           (if (and (not (uiop:emptyp string))
+                    (every #'digit-char-p string))
+               (parse-integer string)
+               string)))
+    (let* ((name (namestring pointer-pathname))
+           (end (when (uiop:emptyp name)
+                  ;; Case of "/" pointer---empty string key.
+                  (list ""))))
+      (with-input-from-string (s name)
+        (append (loop for char = (read-char s nil nil)
```

Comment:
Can't you use `position` and `subseq` instead of `loop`?

Reply:
I can't wrap my head about how to make it cleanly and functionally, so loop seems more readable to me :D

And then, it's a pain to work with subsequences in pure CL without libs, so maybe working with raw chars is a reasonable approach.